webui/react/src/hooks/useModal/useModalExperimentCreate.tsx

```diff
@@ -246,26 +198,78 @@ const useModalExperimentCreate = (props?: Props): ModalHooks => {
       }
 
       setModalState(prev => ({ ...prev, error: errorMessage }));
+
+      // We throw an error to prevent the modal from closing.
+      throw new Error();
```

Comment:
suggestion: this seems to get swallowed by antd modal and not show up anywhere but in anycase it might help if we still include the original error or a message for it.